File: scenarios/scenario_tests/src/test/java/intellij_generate/scenarios/S04TransitiveWithExportTest.java
package intellij_generate.scenarios;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static intellij_generate.scenarios.TestUtil.loadBazelGeneratedImlFile;
import static intellij_generate.scenarios.TestUtil.removeWorkingDirectory;
import static intellij_generate.scenarios.TestUtil.xpathList;
import static java.util.Arrays.asList;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class S04TransitiveWithExportTest {
  private static String grandparentImlContent;
  private static String parentImlContent;
  private static String childImlContent;

  @BeforeAll
  public static void before_all() {
    grandparentImlContent = loadBazelGeneratedImlFile("04_transitive_via_export/grandparent/idea_grandparent_module.iml");
    parentImlContent = loadBazelGeneratedImlFile("04_transitive_via_export/parent/idea_parent_module.iml");
    childImlContent = loadBazelGeneratedImlFile("04_transitive_via_export/child/idea_child_module.iml");
  }

  @Test
  public void sample_jar_dependencies() {
    assertEquals(asList("external/com_google_guava_guava/jar/guava-19.0.jar!/"),
      removeWorkingDirectory(
        xpathList(grandparentImlContent, "/module/component/orderEntry[@type='module-library' and not(@scope)]/library/CLASSES/root/@url")));

    assertEquals(asList("external/com_google_guava_guava/jar/guava-19.0.jar!/"),
      removeWorkingDirectory(
        xpathList(parentImlContent, "/module/component/orderEntry[@type='module-library' and not(@scope)]/library/CLASSES/root/@url")));

    assertEquals(asList("external/com_google_guava_guava/jar/guava-19.0.jar!/"),
      removeWorkingDirectory(
        xpathList(childImlContent, "/module/component/orderEntry[@type='module-library' and not(@scope)]/library/CLASSES/root/@url")));
  }
}

File: scenarios/scenario_tests/pytest/s04_transitive_with_export_test.py
import unittest
from scenario_test_support import *

class S04TransitiveWithExportTest(unittest.TestCase):

    archive = load_archive("04_transitive_via_export/intellij_files")

    grandparent_iml_content = archive["04_transitive_via_export/grandparent/grandparent.iml"]
    parent_iml_content = archive["04_transitive_via_export/parent/parent.iml"]
    child_iml_content = archive["04_transitive_via_export/child/child.iml"]

    def test_compile_deps_dont_show_up_in_test_deps(self):
        self.assertEqual(junit5_jars(), find_all_test_jar_libraries(self.child_iml_content))

    def test_sample_jar_dependencies(self):
        self.assertEqual([
            "jar://${BAZEL_INFO_EXECUTION_ROOT}/external/com_google_guava_guava/jar/guava-19.0.jar!/"
        ], find_all_plain_jar_libraries(self.grandparent_iml_content))

        self.assertEqual([
            "jar://${BAZEL_INFO_EXECUTION_ROOT}/external/com_google_guava_guava/jar/guava-19.0.jar!/"
        ], find_all_plain_jar_libraries(self.parent_iml_content))

        self.assertEqual([
            "jar://${BAZEL_INFO_EXECUTION_ROOT}/external/com_google_guava_guava/jar/guava-19.0.jar!/"
        ], find_all_plain_jar_libraries(self.child_iml_content))
